drivers/volume/common.go

```diff
@@ -0,0 +1,212 @@
+func (d *DefaultDriver) RefreshDriverEndpoints() error {
+	return nil
+}
+
+// CleanupVolume forcefully unmounts/detaches and deletes a storage volume.
+// This is only called by Torpedo during cleanup operations, it is not
+// used during orchestration simulations.
+func (d *DefaultDriver) CleanupVolume(name string) error {
+	return nil
+}
+
+// GetStorageDevices returns the list of storage devices used by the given node.
+func (d *DefaultDriver) GetStorageDevices(n node.Node) ([]string, error) {
+	//TODO: Implement
+	devPaths := []string{}
+	return devPaths, nil
+}
+
+// RecoverDriver will recover a volume driver from a failure/storage down state.
+// This could be used by a volume driver to recover itself from any underlying storage
+// failure.
+func (d *DefaultDriver) RecoverDriver(n node.Node) error {
+	return nil
+}
+
+// ValidateCreateVolume validates whether a volume has been created properly.
+// params are the custom volume options passed when creating the volume.
+func (d *DefaultDriver) ValidateCreateVolume(name string, params map[string]string) error {
+	return nil
+}
+
+// ValidateUpdateVolume validates if volume changes has been applied
+func (d *DefaultDriver) ValidateUpdateVolume(vol *Volume, params map[string]string) error {
+	return nil
+}
+
+// ValidateDeleteVolume validates whether a volume is cleanly removed from the volume driver
+func (d *DefaultDriver) ValidateDeleteVolume(vol *Volume) error {
+	return nil
+}
+
+// ValidateVolumeCleanup checks if the necessary cleanup has happened for the volumes by this driver
+func (d *DefaultDriver) ValidateVolumeCleanup() error {
+	return nil
+}
+
+// ValidateVolumeSetup validates if the given volume is setup correctly in the cluster
+func (d *DefaultDriver) ValidateVolumeSetup(vol *Volume) error {
+	return nil
+}
+
+// StopDriver must cause the volume driver to exit on a given node. If force==true, the volume driver should get killed ungracefully
+func (d *DefaultDriver) StopDriver(nodes []node.Node, force bool) error {
+	return nil
+}
+
+// GetNodeForVolume returns the node on which the volume is attached
+func (d *DefaultDriver) GetNodeForVolume(vol *Volume, timeout time.Duration, retryInterval time.Duration) (*node.Node, error) {
+	return nil, nil
+}
+
+// ExtractVolumeInfo extracts the volume params from the given string
+func (d *DefaultDriver) ExtractVolumeInfo(params string) (string, map[string]string, error) {
+	var volParams map[string]string
+	return "", volParams, nil
+}
+
+// RandomizeVolumeName randomizes the volume name from the given name
+func (d *DefaultDriver) RandomizeVolumeName(params string) string {
+	re := regexp.MustCompile("(" + api.Name + "=)([0-9A-Za-z_-]+)(,)?")
+	return re.ReplaceAllString(params, "${1}${2}_"+uuid.New()+"${3}")
+}
+
+// WaitDriverUpOnNode must wait till the volume driver becomes usable on a given node
+func (d *DefaultDriver) WaitDriverUpOnNode(n node.Node, timeout time.Duration) error {
+	return nil
```

Comment:
These should return NotImplented/NotSupported error

Reply:
Returning NotSupported type error objects.